core/src/main/java/com/linecorp/armeria/common/multipart/BodyPart.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Copyright (c) 2020 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linecorp.armeria.common.multipart;
+
+import javax.annotation.Nullable;
+
+import org.reactivestreams.Publisher;
+
+import com.linecorp.armeria.common.HttpData;
+
+/**
+ * A body part entity.
+ */
+public interface BodyPart {
+
+    // Forked from https://github.com/oracle/helidon/blob/ab23ce10cb55043e5e4beea1037a65bb8968354b/media
+    // /multipart/src/main/java/io/helidon/media/multipart/BodyPart.java
+
+    /**
+     * Returns a new {@link BodyPartBuilder}.
+     */
+    static BodyPartBuilder builder() {
+        return new BodyPartBuilder();
+    }
+
+    /**
+     * Returns the reactive representation of the part content.
+     */
+    Publisher<HttpData> content();
```

Comment:
I'm curious this should return `Publisher<? extends HttpData>`. :thinking:

Reply:
It is possible. I've tried to return `Publisher<? extends HttpData>`.
Then users should assign a local variable with `? extends`. 
```java
Publisher<? extends HttpData> content = bodyPart.content();
```

I agree to take `Publisher<? extends HttpData>`, but not sure that it is useful to return `Publisher<? extends HttpData>`.

Reply:
Agreed. Better returning `Publisher<HttpData>` for simplicity.